arch/x86/core/acpi.c

```diff
@@ -34,7 +34,8 @@ static bool check_sum(struct acpi_sdt *t)
 	for (int i = 0; i < t->length; i++) {
 		sum += p[i];
 	}
-	return sum == 0;
+
+	return !!(sum == 0);
```

Comment:
This looks like a needless noop to me. "`sum == 0`" has boolean type already, so double-negating it produces an identity function.

Reply:
same, I don't understand this change.